### PR TITLE
Tala: Perform developer checklist, app-level code review

### PR DESCRIPTION
**Pull request by mmomin-crest**
The Integration Tests and Static Tests are Failing in the pipeline. which will be resolved when the app listing and expected failures MRs will get merged respectively. Hence, this MR can be reviewed.

App listing MR:  [!9874](https://cd.splunkdev.com/phantom/phantom/-/merge_requests/9874)
Expected failure MR: [!1641](https://cd.splunkdev.com/phantom/app_release/-/merge_requests/1641)

#### Please check if your PR fulfills the following requirements:
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
- [x] Limit your pull request to one application and submit multiple pull requests if necessary.
#### Please check the type of change your PR introduces:
- [x] Code style update (formatting, renaming)

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

---
Thanks for contributing!

### CODE REVIEW

File: tala_connector.py [hunk 1]
```diff
@@ -774,11 +792,11 @@ def finalize(self):
...
             headers['Referer'] = 'https://127.0.0.1/login'
 
-            print ("Logging into Platform to get the session id")
+            print("Logging into Platform to get the session id")
             r2 = requests.post("https://127.0.0.1/login", verify=False, data=data, headers=headers)
             session_id = r2.cookies['sessionid']
         except Exception as e:
-            print ("Unable to get session id from the platfrom. Error: " + str(e))
+            print("Unable to get session id from the platfrom. Error: " + str(e))
             exit(1)
```
Comment (ngaglani-crest) [inline tala_connector.py]: Replace exit() with sys.exit() wherever applicable.

File: tala.json [hunk 2]
```diff
@@ -1054,24 +1051,12 @@
                         "tala-cloud"
                     ],
                     "column_name": "Report Uri",
```
Comment (ngaglani-crest) [inline tala.json]: This should be `Report URI`

File: tala_connector.py [hunk 3]
```diff
@@ -262,9 +265,9 @@ def _handle_create_project(self, param):
         summary = action_result.update_summary({})
         summary['project_id'] = response['id']
 
-        # Return success, no need to set the message, only the status
+        # Return success status with appropriate message
         # BaseConnector will create a textual message based off of the summary dictionary
-        return action_result.set_status(phantom.APP_SUCCESS)
+        return action_result.set_status(phantom.APP_SUCCESS, TALA_CREATED_PROJECT_SUCC)
```
Comment (ngaglani-crest) [inline tala_connector.py]: Please revert this change. We are setting `project_id` in the summary, hence, action_resilt's message will not be empty.

File: tala_connector.py [hunk 4]
```diff
@@ -472,8 +477,12 @@ def _handle_create_scan(self, param):
         project_ids = param.get('project_ids', None)
 
         try:
-            project_ids_list = [int(x) for x in project_ids.split(",")]
-        except:
+            project_ids_list = []
+            for x in project_ids.split(","):
+                if int(x) < 0:
```
Comment (ngaglani-crest) [inline tala_connector.py]: Are we sure that project IDs should be > 0?

File: tala_connector.py [hunk 5]
```diff
@@ -497,9 +506,9 @@ def _handle_create_scan(self, param):
         summary = action_result.update_summary({})
         summary['message'] = response['message']
 
-        # Return success, no need to set the message, only the status
+        # Return success status with appropriate message
         # BaseConnector will create a textual message based off of the summary dictionary
-        return action_result.set_status(phantom.APP_SUCCESS)
+        return action_result.set_status(phantom.APP_SUCCESS, TALA_CREATED_SCAN_SUCC)
```
Comment (ngaglani-crest) [inline tala_connector.py]: Please revert this. Same as above.

File: tala_connector.py [hunk 6]
```diff
@@ -553,9 +562,9 @@ def _handle_get_status(self, param):
         summary = action_result.update_summary({})
         summary['num_projects'] = len(response)
 
-        # Return success, no need to set the message, only the status
+        # Return success status with appropriate message
         # BaseConnector will create a textual message based off of the summary dictionary
-        return action_result.set_status(phantom.APP_SUCCESS)
+        return action_result.set_status(phantom.APP_SUCCESS, TALA_GET_SCAN_STATUS_SUCC)
```
Comment (ngaglani-crest) [inline tala_connector.py]: Ditto

File: tala_connector.py [hunk 7]
```diff
@@ -597,8 +606,12 @@ def _handle_download_policy_bundle(self, param):
 
         project_id = param['project_ids']
         try:
-            [int(x) for x in project_id.split(",")]
-        except:
+            project_id_formatted = []
+            for x in project_id.split(","):
+                if int(x) < 0:
```
Comment (ngaglani-crest) [inline tala_connector.py]: Same as above.

File: tala_connector.py [hunk 8]
```diff
@@ -646,8 +659,12 @@ def _handle_synchronize_projects(self, param):
         scan_id = param['scan_id']
         project_ids = param['project_ids']
         try:
-            project_ids_formatted = [int(x) for x in project_ids.split(",")]
-        except:
+            project_ids_formatted = []
+            for x in project_ids.split(","):
+                if int(x) < 0:
```
Comment (ngaglani-crest) [inline tala_connector.py]: Ditto.

File: tala_connector.py [hunk 9]
```diff
@@ -753,15 +771,15 @@ def finalize(self):
     username = args.username
     password = args.password
```
Comment (ngaglani-crest) [inline tala_connector.py:784]: Add a new command line argument as `verify`. Refer IP Control app for the same.

File: tala_connector.py [hunk 10]
```diff
@@ -774,11 +792,11 @@ def finalize(self):
             headers['Cookie'] = 'csrftoken=' + csrftoken
             headers['Referer'] = 'https://127.0.0.1/login'
 
-            print ("Logging into Platform to get the session id")
+            print("Logging into Platform to get the session id")
             r2 = requests.post("https://127.0.0.1/login", verify=False, data=data, headers=headers)
```
Comment (ngaglani-crest) [inline tala_connector.py]: Replace 127.0.0.1 with the appropriate BaseConnector method, wherever applicable.

File: tala_connector.py [hunk 11]
```diff
@@ -68,7 +69,7 @@ def _process_html_response(self, response, action_result):
             split_lines = error_text.split('\n')
```
Comment (ngaglani-crest) [inline tala_connector.py:73]: Remove the script, style, footer and navigation part from the HTML message

File: tala_connector.py [hunk 12]
```diff
@@ -370,9 +373,9 @@ def _handle_list_projects(self, param):
         for item in data:
             action_result.add_data(data[item])
 
-        # Return success, no need to set the message, only the status
+        # Return success status with appropriate message
         # BaseConnector will create a textual message based off of the summary dictionary
-        return action_result.set_status(phantom.APP_SUCCESS)
+        return action_result.set_status(phantom.APP_SUCCESS, TALA_LIST_PROJECTS_SUCC)
```
Comment (ngaglani-crest) [inline tala_connector.py]: Please revert this. We are setting `Num Projects` in summary.

File: tala_connector.py (hunk 1 above)
Comment (mmomin-crest) [reply tala_connector.py]: Done.

File: tala.json (hunk 2 above)
Comment (mmomin-crest) [reply tala.json]: Changed.

File: tala_connector.py (hunk 4 above)
Comment (mmomin-crest) [reply tala_connector.py]: Reverted back as I do not get a proper document about project id validation.

File: tala_connector.py (hunk 3 above)
Comment (mmomin-crest) [reply tala_connector.py]: Reverted.

File: tala_connector.py (hunk 5 above)
Comment (mmomin-crest) [reply tala_connector.py]: Reverted.

File: tala_connector.py (hunk 6 above)
Comment (mmomin-crest) [reply tala_connector.py]: Reverted.

File: tala_connector.py (hunk 7 above)
Comment (mmomin-crest) [reply tala_connector.py]: Reverted back as I do not get a proper document about project id validation.

File: tala_connector.py (hunk 8 above)
Comment (mmomin-crest) [reply tala_connector.py]: Reverted back as I do not get a proper document about project id validation.

File: tala_connector.py (hunk 9 above)
Comment (mmomin-crest) [reply tala_connector.py:784]: Added command-line argument as verify.

File: tala_connector.py (hunk 10 above)
Comment (mmomin-crest) [reply tala_connector.py]: Replaced with appropriate BaseConnector method.

File: tala_connector.py (hunk 12 above)
Comment (mmomin-crest) [reply tala_connector.py]: Reverted.

File: tala_connector.py (hunk 11 above)
Comment (mmomin-crest) [reply tala_connector.py:73]: Added code for removing the script, style, footer, and navigation part from the HTML message.

File: tala_connector.py [hunk 13]
```diff
@@ -210,9 +211,20 @@ def _download_file_to_vault(self, action_result, endpoint, json, file_name):
 
         if success:
             action_result.set_status(phantom.APP_SUCCESS, "Transferred file")
+            # Check if file with same file name and size is available in vault and save only if it is not available
+            try:
+                _, _, data = ph_rules.vault_info(container_id=self.get_container_id(), vault_id=new_vault_id, file_name=file_name)
+                if not data:
+                    _, _, data = ph_rules.vault_info(vault_id=new_vault_id)
```
Comment (ngaglani-crest) [inline tala_connector.py:218]: This check is not required.

Comment (ngaglani-crest) [inline tala_connector.py]: This check is not required.

Comment (mmomin-crest) [reply tala_connector.py]: Deleted extra check.

**Review by ngaglani-crest — DISMISSED** [review]

**Review by gary-phantom — DISMISSED** [review]

File: release_notes/unreleased.md [hunk 14]
```diff
@@ -1 +1,2 @@
 **Unreleased**
+* Releasing as open-source Community app [PAPP-21787]
```
Comment (ngaglani-crest) [inline release_notes/unreleased.md]:
```suggestion
* Releasing as a community-supported app [PAPP-21787]
```

Comment (mmomin-crest) [reply release_notes/unreleased.md]: Updated.

**Review by ngaglani-crest — APPROVED** [review]
